crates/ruff_python_parser/src/lexer.rs

```diff
@@ -994,6 +994,21 @@ impl<'src> Lexer<'src> {
                     )));
                 };
 
+                // The lexer might've eaten some whitespaces to calculate the `indentation`. For
+                // example:
+                //
+                // ```py
+                // if first:
+                //     if second:
+                //         pass
+                //     foo
+                // #   ^
+                // ```
+                //
+                // Here, the cursor is at `^` and the `indentation` contains the whitespaces before
+                // the `pass` token.
+                self.cursor.start_token();
```

Comment:
I must admit that reading `self.cursor.start_token()` is hard to understand. Why are we starting a token when we're just about to return it. Your comment certainly helps. That makes me wonder if we can instead add a method on `Lexer` with a more meaningful name. 

```rust
self.update_token_start_to_current_offset() // A bi tlong
self.reset_token_start()
```

but I must admit, I don't find them much better than what we have. But maybe you have an idea for a better name?

Reply:
Yeah, I agree. Let me think about this.